### DISCUSSION TITLE
Google Keep Import

### DISCUSSION THREAD

**Discussion by mrsdoubtfire613** [usememos/memos#3795] [Ideas]
Hello!

Is it possible to - or are there plans to add the functionality of - import Google Keep notes in bulk?  I've got many dozen (potentially more) keep notes that I'd like to bring over but doing some manually would be extremely time-consuming.  Maybe a way to take Keep notes from Google Takeout?  I saw #1705 but it links to a dead project that doesn't appear to work anymore.

Thank you!

**Comment by vld333** [comment]
Also would love to see this. I have hundreds of items in my google keep.

**Reply by terbo** [reply]
It's a WIP but I fixed the script, though there are a few more updates needed,
 and some potential problems - it currently doesn't import images, and it doesn't read HTML notes.

I suggest that you backup your memos.db, then create a new user and import notes to that user.
That, or run another version of memos on another host or port and use that to test this out.

I haven't figured out how to mass delete notes from a specific user and I had to do that with sqlite3 while testing this,
so YMMV.

Hopefully I'll get some time to update this and make sure everything works, but I imported most of my 1400 notes with titles, and labels converted to tags, so it works pretty well for me at the moment, though I haven't merged it with my main memo database yet.

Gist: https://gist.github.com/terbo/f87a137a13af3d12806795c19fd796d8

https://github.com/erikelisath/keep-to-markdown - for converting keep notes
https://github.com/hyoban/kirika - now defunct, but used to convert memo formats
Also https://github.com/usememos/memos/issues/4638 may be a better solution.

**Comment by barelias** [comment]
Hey guys, I also did a gist to import from the Keep Takeout, this one import images.

https://gist.github.com/barelias/a9232d85339cf6b8b0b0f650545d0a21

**Comment by joshefin** [comment]
One more option to try - https://github.com/joshefin/keep2memos
